File: DBObjTest/DBTest/Models/Runner+Helpers.m
//
//  Runner+Helpers.m
//  DBTest
//
//  Created by kevin delord on 01/10/14.
//  Copyright (c) 2014 kevin delord. All rights reserved.
//

#import "Runner+Helpers.h"

@implementation Runner (Helpers)

- (NSString *)description {
    return [NSString stringWithFormat:@"%@ : %@ - %@", self.objectID.URIRepresentation.lastPathComponent, self.name, self.position];
}

- (id)uniqueIdentifier {
    return self.objectID;
}

- (void)updateWithDictionary:(NSDictionary *)dict {
    self.name = GET_STRING(dict, @"name");
    self.position = GET_NUMBER(dict, @"position");
}

- (NSString *)invalidReason {
    return nil;
}

- (void)deleteChildEntities {
    [super deleteChildEntities];
}

- (void)saveEntityAsNotDeprecated {
    [super saveEntityAsNotDeprecated];
}

+ (BOOL)verbose {
    return true;
}

+ (NSString *)sortingAttributeName {
      return @"position";
}

+ (NSPredicate *)primaryPredicateWithDictionary:(NSDictionary *)dictionary {
    return [NSPredicate predicateWithFormat:@"name ==[c] %@", GET_STRING(dictionary, @"name")];
}

- (BOOL)shouldUpdateEntityWithDictionary:(NSDictionary *)dictionary {
    return true;
}

+ (void)countEntity {
    NSLog(@"%@", Runner.MR_findAll);
}

@end

File: Example/DBSwiftTest/Runner+Helpers.swift
//
//  Runner+Helpers.swift
//  DBSwiftTest
//
//  Created by kevin delord on 01/10/14.
//  Copyright (c) 2014 Smart Mobile Factory. All rights reserved.
//

import Foundation

extension Runner {

	public override var description : String {
        return "\(self.objectID.URIRepresentation().lastPathComponent) : \(self.name) - \(self.position)"
    }

    override public func uniqueIdentifier() -> AnyObject {
        return self.objectID;
    }

	public override func saveEntityAsNotDeprecated() {
		super.saveEntityAsNotDeprecated()
	}


	public override func updateWithDictionary(dictionary: [NSObject : AnyObject]?, inContext savingContext: NSManagedObjectContext) {
        self.name = GET_STRING(dictionary, "name")
        self.position = GET_NUMBER(dictionary, "position")
    }

    override public func invalidReason() -> String? {
        return nil
    }


	public override func deleteChildEntitiesInContext(context: NSManagedObjectContext) {
		//
	}

    override public class func verbose() -> Bool {
        return true
    }

    override public class func sortingAttributeName() -> String? {
        return "position"
    }

    override public class func primaryPredicateWithDictionary(dictionary: [NSObject:AnyObject]?) -> NSPredicate? {
        // if return nil will only take the first entity created (if any) and update it. By doing so only ONE entity will ever be created.
        // otherwise use the one find bu the predicate.
		if let name = GET_STRING(dictionary, "name") {
        	return NSPredicate(format: "name ==[c] %@", name)
		}
		return nil
    }


	public override func shouldUpdateEntityWithDictionary(dictionary: [NSObject : AnyObject]?, inContext savingContext: NSManagedObjectContext) -> Bool {
        return true
    }

    class func countEntityInContext(context: NSManagedObjectContext) {
        print(Runner.MR_findAllInContext(context))
    }
}